tests/unit/model/errors-attribute-test.js

```diff
@@ -89,6 +95,25 @@ module('unit/model/errors-attribute', function (hooks) {
     );
   });
 
+  test('schema with flag set to true causes _removeError to be a noop', function (assert) {
+    this.owner.register('service:m3-schema', TestSchemaFlagOn);
+
+    let model = run(() => {
+      return this.store.push({
+        data: {
+          id: 'urn:book:1',
+          type: 'com.example.bookstore.Book',
+          attributes: {
+            author: ['urn:author:1'],
+          },
+        },
+      });
+    });
+    const getErrorsSpy = this.sinon.spy(model, 'errors', ['get']);
+    model._removeError('author');
```

Comment:
These kinds of tests couple us to current specific implementation. If we refactor `_removeError` to a different method, this test will fail, even though the behavior might still be correct. Can we write this test to assert user observable behavior?

Reply:
Ended changing to use `model.setUnknownProperty('title', 'Harry Potter');` which calls `_removeError`

Reply:
It seems like you are still testing the internal implementation though. The user visible behavior here, is that if `useUnderlyingErrorsValue` is `true` then you can access `errors` after setting a property. If it's false, and you set a prop, the error gets removed, right?

Reply:
Lets also create the errors using the public api and not by setting the property directly.  The `save errors getting updated via the store and removed upon setting a new value` has an example of how to get invalid errors. I've cleaned up that test as the setup wasn't quite correct here: https://github.com/hjdivad/ember-m3/pull/1251